tcms/core/tests/test_sendtestemail.py

```diff
@@ -0,0 +1,13 @@
+from django.core.management import call_command
+from django.test import TestCase, override_settings
+import socket
+
+
+class TestSendTestEmail(TestCase):
+    """Test manage.py sendtestemail command"""
+
+    @override_settings(EMAIL_HOST='bogus.nonexistent')
+    def test_send_false_server(self):
+        """test command with fake SMTP server"""
+        call_command('sendtestemail', '--admins')
+        self.assertRaises(socket.gaierror)
```

Comment:
That should be 
```
with self.assertRaises(...):
    call_command()
```

however even modifying my own settings file in devel mode doesn't cause this command to crash & report traceback.

IOW this is passing b/c it is always passing.

Reply:
I get the socker.gaierror only if I try to send to a specific address, so maybe the test and the docs should default to that usage scenario.

Reply:
The error occurs if the SMTP server name cannot be resolved. So I use EMAIL_HOST='bogus.nonexistent' - it should never be correctly resolved. In case of invalid e-mail, the system waits for SMTP response before reaction. If we have invalid server IP throws timeout but after a very long time.

Reply:
But this does not work this way during test. With:
`with self.assertRaises(...):
    call_command()`
the exception is not raised:
`with self.assertRaises(...):
    call_command()`
 My original test always passes, even without `@override settings`

Reply:
your current assertion is wrong, see my previous comment. it tells you `AssertionError: gaierror not raised` b/c assertRaises expects it's body to raise the specific exception specified as argument but that never happens b/c the body is empty. 

When unsure consult the documentation:
https://docs.python.org/3/library/unittest.html#unittest.TestCase.assertRaises